crates/stages/src/stages/bodies.rs

```diff
@@ -50,16 +53,21 @@ pub(crate) const BODIES: StageId = StageId("Bodies");
 /// - The [`CumulativeTxCount`][reth_interfaces::db::tables::CumulativeTxCount] table
 /// - The [`Transactions`][reth_interfaces::db::tables::Transactions] table
 /// - The [`TransactionHashNumber`][reth_interfaces::db::tables::TransactionHashNumber] table
-#[derive(Debug)]
-pub struct BodyStage<D: BodyDownloader, C: Consensus> {
+pub struct BodyStage<D: BodyDownloader> {
     /// The body downloader.
     pub downloader: D,
     /// The consensus engine.
-    pub consensus: Arc<C>,
+    pub consensus: Arc<dyn Consensus>,
+}
+
+impl<D: BodyDownloader + Debug> Debug for BodyStage<D> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("BodyStage").field("downloader", &self.downloader).finish()
```

Comment:
this should at least be
```suggestion
        f.debug_struct("BodyStage").field("downloader", &self.downloader).finish_non_exhaustive()
```
in general, i don't like that we handroll debug impls now. why not just mark `trait Consensus: Debug`?